client/app/Streamfinder.jsx

```diff
@@ -26,40 +26,64 @@ import MediaDetail from '../features/media/MediaDetail';
 import Account from '../features/accountPage/Account';
 import ErrorBoundary from '../features/sharedComponents/ErrorBoundary';
 import './Streamfinder.css';
-/* the idea:
-upon auth being valid -
-* auth will send username and current history ids (from user schema) to StreamerFinder
-  and state will bet set
-* Each component can use component did update for when their props
-change.  upon this happening they can make an ajax request to server
-to pull data needed from MovieSchema
-*/
+
 class Streamfinder extends React.Component {
   constructor(props) {
     super(props);
 
+    this.sessionExpired = this.sessionExpired.bind(this);
     this.updateSession = this.updateSession.bind(this);
     this.checkCache = this.checkCache.bind(this);
     this.updateCache = this.updateCache.bind(this);
-    this.handleSearchIdSwitch = this.handleSearchIdSwitch.bind(this);
 
     this.cache = new Map();
     this.sessionToken = this.props.sessionToken || null;
 
     this.state = {
       sessionToken: this.props.sessionToken || null,
-      //user establish because this is prior to auth being hooked up
-      //this is under the impression auth was valid and currentId was sent
-      //to this component and updated via component did update.
-      // currentId: 10138,
-      user: 'lil timmy'
     };
   }
 
-  updateSession(token) {
-    console.log(token);
-    window.localStorage.setItem('sessionToken', 'fakehash');
-    this.setState({ sessionToken: token });
+  updateSession(username) {
+    if (!username) {
+      window.localStorage.removeItem('sessionToken');
+      this.setState({ sessionToken: null });
+    } else {
+      const sessionToken = btoa(`{"username": "${username}", "date": ${new Date().getTime()}}`);
+      window.localStorage.setItem('sessionToken', sessionToken);
+      this.setState({ sessionToken });
+      return sessionToken;
+    }
+  }
+
+  sessionExpired() {
+    // Checks expiration date decoded from sessionToken
+    if (this.state.sessionToken) {
+      const expiredTime = (new Date().getTime() - JSON.parse(atob(this.state.sessionToken)).date) / 1000;
+      if (expiredTime > 7776000) {
+        alert('Session expired. Please sign in to continue.');
+        return true;
+      } else {
+        let expiration;
+        const daysRemaining = Number.parseFloat((7776000 - expiredTime) / 86400).toPrecision(2);
+        if (daysRemaining < 1) {
+          expiration = `less than ${Number.parseFloat(daysRemaining * 24).toPrecision(2)} hours`;
+        } else {
+          expiration = daysRemaining + ' days';
+        }
+        console.log(`Session expires in ${expiration}`);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  currentUser(sessionToken = window.localStorage.getItem('sessionToken')) {
+    // Returns username decoded from sessionToken
+    if (sessionToken) {
+      const username = JSON.parse(atob(sessionToken)).username;
```

Comment:
What does "atob" stand for?

Reply:
I don't know. It's native JS though. I think it might just be A to B. It's a simple encryption algorithm. Its opposite is `atob()`